compaction/compaction_manager.cc

```diff
@@ -2209,7 +2205,7 @@ compaction_backlog_manager::~compaction_backlog_manager() {
 
 void compaction_manager::register_backlog_tracker(table_state& t, compaction_backlog_tracker new_backlog_tracker) {
     auto& cs = get_compaction_state(&t);
-    cs.backlog_tracker = std::move(new_backlog_tracker);
+    cs.backlog_tracker.transfer_backlog(std::move(new_backlog_tracker));
     register_backlog_tracker(cs.backlog_tracker);
```

Comment:
what if we make backlog tracker optional and just rely on compaction_backlog_tracker's move ctor?

then we can delete move_assignment altogether
```

    compaction_state cs;

    compaction_backlog_tracker new_backlog_tracker;

    cs.backlog_tracker.emplace(std::move(new_backlog_tracker)); // this invokes move ctor.
```

Reply:
I think that should work